core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -1490,7 +1491,9 @@ public String name()
     @Override
     public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
     {
-      final String arg = args.get(0).eval(bindings).asString();
+      final ExprEval eval = args.get(0).eval(bindings);
+      assert !(eval instanceof ArrayExprEval);
```

Comment:
> Besides that, since this error could happen due to legitimate user input, it should have a nicer error message.

Oh yeah, that's true. I forgot about that even though that was the original issue. I changed from throwing an exception to an assert here since it should never happen on the query side and `apply()` will be called per row. A systematic way to check before calling `apply()` makes most sense to me.

> Also, is there a reason to only adjust strlen in this patch? Lots of other functions have similar issues. I'm wondering if we should split this part off into a separate patch, and make sure we address it systematically.

There is no reason to fix only `strlen`. I just happened to find it while adding unit tests and thought fixed it. I reverted this change back for now.